process/block/alteredaccounts/alteredAccountsProvider.go

```diff
@@ -139,50 +126,29 @@ func (aap *alteredAccountsProvider) processMarkedAccountData(
 		Nonce:   userAccount.GetNonce(),
 	}
 
-	if len(markedAccountTokens) > 0 {
-		for _, tokenData := range markedAccountTokens {
-			err = aap.fetchTokensDataForMarkedAccount(encodedAddress, tokenData, alteredAccounts)
-			if err != nil {
-				log.Warn("cannot fetch token data for marked account", "error", err)
-				return err
-			}
+	for tokenKey, tokenData := range markedAccountTokens {
+		err = aap.fetchTokensDataForMarkedAccount([]byte(tokenKey), encodedAddress, userAccount, tokenData, alteredAccounts)
```

Comment:
This function should **only** do what the name says: fetch tokens data for an account.
It should return `&indexer.AccountTokenData`, then here you can add it to your altered accounts.

Reply:
renamed to `addTokensDataForMarkedAccount` 
I wouldn't return the AccountTokenData as the account the function is adding to already exists in the map, so I think it is better to update the map inside the function, instead of returning it and adding to the map where I've just read it from (inside the function).

Reply:
Agree